futurex_openedx_extensions/helpers/tasks_utils.py

```diff
@@ -139,11 +141,13 @@ def _generate_csv_with_tracked_progress(
                     # update task progress
                     fx_task.progress = progress
                     fx_task.save()
-        storage_path = _upload_file_to_storage(tmp_file.name, filename)
+        tenant = TenantConfig.objects.get(id=fx_task.tenant_id)
+        storage_path = _upload_file_to_storage(tmp_file.name, filename, tenant.external_key)
```

Comment:
we always use `tenant.id`. The external key id for integration with external services
```suggestion
        storage_path = _upload_file_to_storage(tmp_file.name, filename, fx_task.tenant_id)
```